app/models/auth_token.rb

```diff
@@ -67,13 +67,10 @@ def self.new_client(token, options = {})
   def self.new_v4_client(token)
     token ||= AuthToken.token
     http_adapter = GraphQL::Client::HTTP.new("https://api.github.com/graphql") do
-      @token = token
-
+      @@token = token # rubocop: disable Style/ClassVars
       # rubocop: disable Lint/NestedMethodDefinition:
       def headers(_context)
-        {
-          "Authorization" => "Bearer #{@token}",
-        }
+        { "Authorization" => "bearer #{@@token}" }
```

Comment:
One way to avoid needing a class variable is taking advantage of the fact that `HTTP.new` creates an object that can have other methods attached to it. Something like this may be possible:

```ruby
http_adapter = GraphQL::Client::HTTP.new("https://api.github.com/graphql") do
  attr_accessor :token

  def headers(_context)
    { "Authorization" => "Bearer #{token}" }
  end
end

http_adapter.token = token
```

Reply:
💡  Way too simple for me to see after looking at it as long as I was